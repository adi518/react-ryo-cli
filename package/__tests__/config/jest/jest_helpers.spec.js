const { cloneDeep } = require("lodash");

const { extendJestConfig } = require("../../../config/jest/jest_helpers");

const commonMockArgs = {
  jestConfig: {
    testMatch: [],
    setupFiles: [],
    setupFilesAfterEnv: [],
    moduleNameMapper: {},
    snapshotSerializers: [],
    collectCoverage: null
  }
};

describe("extendJestConfig", () => {
  test("with build (development) script", () => {
    const mockArgs = {
      ...cloneDeep(commonMockArgs)
    };
    const result = extendJestConfig(mockArgs);

    expect(result).toMatchSnapshot();
  });

  test("with build:production script", () => {
    const mockArgs = {
      ...cloneDeep(commonMockArgs),
      env: { production: true }
    };
    const result = extendJestConfig(mockArgs);

    expect(result).toMatchSnapshot();
  });
});
