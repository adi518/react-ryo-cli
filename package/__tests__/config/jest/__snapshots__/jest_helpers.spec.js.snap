// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extendJestConfig with build (development) script 1`] = `
Object {
  "extendedJestConfig": Object {
    "collectCoverage": null,
    "moduleNameMapper": Object {
      "\\\\.(css|less|scss|sss|styl)$": "<rootDir>/node_modules/identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/node_modules/react-build/config/jest/fileMock.js",
    },
    "setupFiles": Array [
      "<rootDir>/node_modules/react-build/config/enzyme/enzyme_setup.js",
    ],
    "setupFilesAfterEnv": Array [
      "<rootDir>/node_modules/react-build/config/jest/jest_setup.js",
    ],
    "snapshotSerializers": Array [
      "enzyme-to-json/serializer",
    ],
    "testMatch": Array [
      "<rootDir>/tests/**/*.{spec,test}.{js,jsx,ts,tsx}",
    ],
  },
}
`;

exports[`extendJestConfig with build:production script 1`] = `
Object {
  "extendedJestConfig": Object {
    "collectCoverage": true,
    "moduleNameMapper": Object {
      "\\\\.(css|less|scss|sss|styl)$": "<rootDir>/node_modules/identity-obj-proxy",
      "\\\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/node_modules/react-build/config/jest/fileMock.js",
    },
    "setupFiles": Array [
      "<rootDir>/node_modules/react-build/config/enzyme/enzyme_setup.js",
    ],
    "setupFilesAfterEnv": Array [
      "<rootDir>/node_modules/react-build/config/jest/jest_setup.js",
    ],
    "snapshotSerializers": Array [
      "enzyme-to-json/serializer",
    ],
    "testMatch": Array [
      "<rootDir>/tests/**/*.{spec,test}.{js,jsx,ts,tsx}",
    ],
  },
}
`;
